### ISSUE TITLE
[REQUEST] Song Organization

### ISSUE THREAD

**Issue by rgnet1** [exislow/tidal-dl-ng#41]
Please allow for every media type (playlist, album, track, etc.) to be able to use the same asset tag. Currently downloading individual tracks won't load the "{album_title}".  I would like all the songs regardles of asset tag to be downloaded in the same folder structure 

{artist_name}/{album_title}/song.extention

This helps keeps things organized

**Comment by exislow** [comment]
The format tags, which are allowed to be used currently are very limited (see https://github.com/exislow/tidal-dl-ng/blob/d5d53c6c921598bfd5f52b6330ddef6c9f7a3ed7/tidal_dl_ng/helper/path.py#L82-L132)

Can you and other user please think about all possibilities you like to have for formatting tracks, albums, playlists etc. and post it here (@rgnet1 you can also just edit your first post) so I can have a look and try to implement it.

You do not need to come up with technical details, just have a look at the actual TIDAL app regarding all the available information for songs etc. and try to describe them here.

**Comment by rgnet1** [comment]
No problem. It would be nice to have 2 features:

1. have  optional global download schema which overrides  all of the download directory formats of the various media types (playlist, artist, album, etc.. )

2. each download schema for media types (playlist, artist, album, etc.. ) should each have access to all the various tags: {album_name}, {artist_name}, {track_number}, {track_name}, etc. Currently if i download an individual track, I cannot use that track's album_name attribute in the download schema. The download literally puts the string "{album_name}" as a folder instead of the actual album. Often times i only download a single song from an album but still need the album name for organization purposes.


I want a global option such that all downloads are organize the same way. this is required for many media organization software like Plex and Jellyfin. Here's the schema directly from plex docs:

`Music/ArtistName/AlbumName/TrackNumber - TrackName.ext`

Here's a practical example, where The Wall album has 2 discs

```
/Music
   /Pink Floyd
      /The Wall
         101 - In the Flesh.mp3
         102 - The Thin Ice.mp3
         201 - Hey You.mp3
         202 - Is There Anybody Out There.mp3
      /Wish You Were Here
         01 - Shine On You Crazy Diamond (Parts I-V).m4a
         02 - Welcome to the Machine.mp3
         03 - Have a Cigar.mp3
   /Foo Fighters
      /One By One
          05 - single song form the album.flac
      /There is Nothing Left to Lose
   /U2
      /Joshua Tree

```

TLDR: Regardless if I download a single track, playlist, or album, I always want the same naming scheme. 

Let me know if this helps give you a better idea of what I'm looking for

**Comment by exislow** [comment]
Currently blocked by https://github.com/tamland/python-tidal/issues/234

**Comment by exislow** [comment]
Done. Will be released soon.

**Comment by rgnet1** [comment]
Thank you for the release. Album name is fixed :) . However, I am noticing the issue is persistent with track_num. Can you take a look at this variable? 

![image](https://github.com/exislow/tidal-dl-ng/assets/12654681/204fc8d1-d1df-49ff-a9f6-e41eb61f1bac)

**Comment by exislow** [comment]
Duplicate https://github.com/exislow/tidal-dl-ng/issues/64

User `{album_track_num}` instead of `{track_num}`

**Comment by rgnet1** [comment]
Oh perfect. Thank you!
